### PR TITLE
Fix handling of exception edges in backward analysis

### LINKED ISSUE #4861: Incorrect logic in `BackwardAnalysisImpl` for exceptional successors
I was looking at the following code:

https://github.com/typetools/checker-framework/blob/b256527a82816115c3df6c120f0f3b0a1f7238a9/dataflow/src/main/java/org/checkerframework/dataflow/analysis/BackwardAnalysisImpl.java#L256-L263

It seems to be trying to check if `block` is an exceptional successor of `pred`.  But the check is wrong, as `((ExceptionBlock) pred).getSuccessor()` will get the non-exceptional successor (`getExceptionalSuccessors()` should be used instead).  Reading the surrounding code, I think the impact of this bug is that in many cases, stores will not get propagated at all along exceptional control-flow edges in the backwards analysis.  Not sure if there are tests for this.

### PR DESCRIPTION
**Pull request by msridhar**
Fixes #4861

### CODE REVIEW

File: dataflow/tests/live-variable/Expected.txt [hunk 1]
```diff
@@ -66,23 +71,75 @@ b   [ LocalVariable ]
...
+0   [ IntegerLiteral ]
+b = 0   [ Assignment ]
+marker (start of try statement #0)   [ Marker ]
+marker (start of try block #0)   [ Marker ]
+f   [ LocalVariable ]
+1   [ IntegerLiteral ]
+a   [ LocalVariable ]
+~~~~~~~~~
+TransferInput#21
+After:   live variables = a, b
```
Comment (msridhar) [inline dataflow/tests/live-variable/Expected.txt:93]: Before the logic change in `BackwardAnalysisImpl`, the live variable set here would incorrectly be missing `b`

**Comment by kelloggm** [comment]
This looks reasonable to me, but I'm not the right person to review it, nor am I sure who is. @mernst, can you either review this change yourself or assign someone else who you think is an appropriate reviewer?

**Review by kelloggm — APPROVED** [review]
LGTM

**Review by mernst — APPROVED** [review]
Thanks!

**Pull request merged**